pkg/cache/sql/store/store.go

```diff
@@ -27,6 +27,15 @@ const (
 		objectnonce BLOB,
 		dekid INTEGER
 	)`
+	// 'key' contains values from the column key of table nodes
+	createLabelsTableFmt = `CREATE TABLE IF NOT EXISTS "%s_labels" (
+		key TEXT NOT NULL REFERENCES "%s"(key) ON DELETE CASCADE,
+		label TEXT NOT NULL,
+		value TEXT NOT NULL,
+		PRIMARY KEY (key, label)
+	)`
+	upsertLabelsStmtFmt = `REPLACE INTO "%s_labels"(key, label, value) VALUES (?, ?, ?)`
+	deleteLabelsStmtFmt = `DELETE FROM "%s_labels" WHERE KEY = ?`
```

Comment:
sql/store/store was designed to be a drop-in replacement to client-go's store type, only backed by SQLite instead of in-memory structures.

It sounds a bit out of place to me to bake labels in here.

Instead, I'd recommend to add it either to ListOptionIndexer or, potentially, a brand new Storage class specifically for labels (altohugh, in that case, logic would still bleed into ListOptionIndexer).

Please take a look at https://github.com/rancher/lasso/blob/master/pkg/cache/sql/informer/indexer.go for inspiration. There Indexer builds upon Store by creating own tables, inserting data into them via RegisterAfterUpsert and then querying in own methods. I think that is similar to what you are achieving here.

Reply:
k, I moved it into the listoption indexer and it seems to fit right.